src/App.tsx

```diff
@@ -1,10 +1,70 @@
+/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { getTodos, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { ErrorMessages } from './types/ErrorMessages';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodoItem } from './components/TodoItem';
+import { StatusFilterOptions } from './types/StatusFilterOptions';
+import { FilterByStatus } from './components/FilterByStatus';
+
+interface GetFilteredTodosFilter {
+  status: StatusFilterOptions;
+}
+
+const getFilteredTodos = (todos: Todo[], filter: GetFilteredTodosFilter) => {
+  let filteredTodos = [...todos];
+
+  if (filter.status !== StatusFilterOptions.All) {
+    filteredTodos = filteredTodos.filter(todo => {
+      return filter.status === StatusFilterOptions.Completed
+        ? todo.completed
+        : !todo.completed;
+    });
+  }
+
+  return filteredTodos;
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todosLoading, setTodosLoading] = useState(true);
```

Comment:
```suggestion
  const [isLoading, setIsLoading] = useState(false);
```
should start with fasle
boolean strats with is has are